### PR TITLE
refactor(cardinal): renaming stuff to make things more clear.

### PR DESCRIPTION
**Pull request by pyrofolium**
Closes: WORLD-722

Rename `Storage` into `GameState`

Instantiation of the ECB was called `NewManager`, renamed to `NewEntityComponentBuffer`

Renamed `IManager` to `IGameStateManager`

It's just not clear what a manager or storage is. This PR is saying Store is a GameState and an EntityComponentBuffer is an aspect of gamestate. 

So now during instantiation of an Engine, it becomes explicitly clear from reading the code that we are passing in an Entity Component Buffer into the engine. 

## Summary by CodeRabbit

- **Refactor**
	- Transitioned from using `StoreManager` to `GameStateManager` to enhance state management.

- **Bug Fixes**
	- Corrected references to state management functions across various tests to ensure consistency.

- **Documentation**
	- Updated documentation to reflect renaming of core components and interfaces.

- **Chores**
	- Adjusted import paths and function calls related to the game state management overhaul.

### CODE REVIEW

File: cardinal/ecs/gamestate/ecb.go [hunk 1]
```diff
@@ -74,7 +74,7 @@
 	return m, nil
 }
 
-func (m *Manager) RegisterComponents(comps []component.ComponentMetadata) error {
+func (m *EntityComponentBuffer) RegisterComponents(comps []component.ComponentMetadata) error {
```
[the remark this answers is not in the document]
Comment (pyrofolium) [reply cardinal/ecs/gamestate/ecb.go:77]: return m.loadArchIDs() returns an error anyway which is the return type of the parent function this is fine.

File: cardinal/ecs/gamestate/ecb.go [hunk 2]
```diff
@@ -172,7 +172,7 @@
 }
 
 // SetComponentForEntity sets the given entity's component data to the given value.
-func (m *Manager) SetComponentForEntity(cType component.ComponentMetadata, id entity.ID, value any) error {
+func (m *EntityComponentBuffer) SetComponentForEntity(cType component.ComponentMetadata, id entity.ID, value any) error {
```
[the remark this answers is not in the document]
Comment (pyrofolium) [reply cardinal/ecs/gamestate/ecb.go]: This is unrelated to this PR. Most likely not even a problem.

**Review by namrapatel — COMMENTED** [review]
What I think we should do:

1. /storage -> /metastore
2. /storage/redis should probably be brought out into /metatore
3. /ecb -> /ecb
4. Manager struct -> EntityCommandBuffer
5. IManager interface only has one implementation, I think it can be deleted and brought back trivially if it is required in the future

**Comment by pyrofolium** [comment]
> What I think we should do:
> 
> 1. `/storage` -> `/metastore`
> 2. `/storage/redis` should probably be brought out into `/metastore`
> 3. `/ecb` -> `/ecb`
> 4. `Manager` struct -> `EntityCommandBuffer`
> 5. `IManager` interface only has one implementation, I think it can be deleted and brought back trivially if it is required in the future

1.,2. This will be another PR. We can discuss it then. I will keep your decision in mind here. 
4. We'll ask Jeremey what the name was. I distinctly remember it was coined by Jeremey "Entity Component Buffer", a play off of ECS. But with there's multiple people making claims on the overall design, so I'm not sure. This isn't too important to me. @jerargus 
5. This one is required I believe. If we are to replace ECB with something else it will be going behind this interface. This interface is what allows us to do this. The likelihood of doing this is quite high.

**Comment by namrapatel** [comment]
> > What I think we should do:
> > 
> > 1. `/storage` -> `/metastore`
> > 2. `/storage/redis` should probably be brought out into `/metastore`
> > 3. `/ecb` -> `/ecb`
> > 4. `Manager` struct -> `EntityCommandBuffer`
> > 5. `IManager` interface only has one implementation, I think it can be deleted and brought back trivially if it is required in the future
> 
> 1.,2. This will be another PR. We can discuss it then. I will keep your decision in mind for that one. 4. We'll ask Jeremey what the name was. I distinctly remember it was coined by Jeremey "Entity Component Buffer", a play off of ECS. But with there's multiple people making claims on the overall design, so I'm not sure. This isn't too important to me. @jerargus 5. This one is required I believe. If we are to replace ECB with something else it will be going behind this interface. This interface is what allows us to do this. The likelihood of doing this is quite high.

Yeah seems fine to do (1) and (2) in another PR if that's how you wanna go about it, but I think if we agree on (3) that should happen in this PR. I don't love the change from ecb -> gamestate, but we should ask others ofc.

File: cardinal/ecs/gamestate/store/iface.go [hunk 3]
```diff
@@ -60,9 +60,9 @@ type TickStorage interface {
 	Recover(txs []message.Message) (*txpool.TxQueue, error)
 }
 
-// IManager represents all the methods required to track Component, Entity, and Archetype information
+// IGameStateManager represents all the methods required to track Component, Entity, and Archetype information
```
Comment (jerargus) [inline cardinal/ecs/gamestate/store/iface.go:63]: I suggesting merging this package with the gamestate package one level above. So there would no longer be a cardinal/ecs/gamestate/store package. Just cardinal/ecs/gamestate.

File: cardinal/ecs/internal/testutil/testutil.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ func InitEngineWithRedis(t *testing.T, s *miniredis.Miniredis) *ecs.Engine {
 		Password: "", // no password set
 		DB:       0,  // use default DB
 	}, Namespace)
-	sm, err := ecb.NewManager(rs.Client)
+	sm, err := gamestate.NewEntityComponentBuffer(rs.Client)
```
Comment (jerargus) [inline cardinal/ecs/internal/testutil/testutil.go:39]: I've always understood this to be an "Entity Command Buffer". It buffers commands (i.e. state changes) related to entities.

Comment (pyrofolium) [reply cardinal/ecs/internal/testutil/testutil.go:39]: alright looks like I hallucinated that memory.

**Comment by pyrofolium** [comment]
> > > What I think we should do:
> > > 
> > > 1. `/storage` -> `/metastore`
> > > 2. `/storage/redis` should probably be brought out into `/metastore`
> > > 3. `/ecb` -> `/ecb`
> > > 4. `Manager` struct -> `EntityCommandBuffer`
> > > 5. `IManager` interface only has one implementation, I think it can be deleted and brought back trivially if it is required in the future
> > 
> > 
> > 1.,2. This will be another PR. We can discuss it then. I will keep your decision in mind for that one. 4. We'll ask Jeremey what the name was. I distinctly remember it was coined by Jeremey "Entity Component Buffer", a play off of ECS. But with there's multiple people making claims on the overall design, so I'm not sure. This isn't too important to me. @jerargus 5. This one is required I believe. If we are to replace ECB with something else it will be going behind this interface. This interface is what allows us to do this. The likelihood of doing this is quite high.
> 
> Yeah seems fine to do (1) and (2) in another PR if that's how you wanna go about it, but I think if we agree on (3) that should happen in this PR. I don't love the change from ecb -> gamestate, but we should ask others ofc.

apologies, wasn't fully getting what you meant by ecb -> ecb so I didn't respond. You meant don't change it. 

I disagree with this. First off ecb is just not a clear name, if nobody told you what ecb was you wouldn't know. Second if we named it to "Entity Command Buffer" That's still not fully clear. GameState is the clearest possible name. The user looks at it, he knows that all code related to the state of the running game. 

We can vote on this if there's no alternative compelling reason to present here, looks like Jeremey cool with the name, it's just @technicallyty, and @zulkhair. @zulkhair you can optionally abstain I guess if you don't have an opinion because you're still learning the code.

**Comment by technicallyty** [comment]
im going to look more into this PR later today, but i do want to voice an opinion that i don't like prefixing interface types with `I`. we have a few interfaces in the code and some are prefixed `I`, some aren't - my vote is to not do it.

**Comment by pyrofolium** [comment]
> im going to look more into this PR later today, but i do want to voice an opinion that i don't like prefixing interface types with `I`. we have a few interfaces in the code and some are prefixed `I`, some aren't - my vote is to not do it for any of 'em.

I'm neutral about this. The reason why the `I` is there is because it was originally called Iface.go. I just renamed it to IGameManager.go without making a decision about the `I`. Doesn't matter, I will remove it.

File: cardinal/ecs/gamestate/ecb.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pyrofolium) [reply cardinal/ecs/gamestate/ecb.go:76]: Can you rewrite everything in rust?

File: cardinal/ecs/gamestate/ecb.go [hunk 5]
```diff
@@ -104,7 +104,7 @@
 }
 
 // RemoveEntity removes the given entity from the ECS data model.
-func (m *Manager) RemoveEntity(idToRemove entity.ID) error {
+func (m *EntityComponentBuffer) RemoveEntity(idToRemove entity.ID) error {
```
[the remark this answers is not in the document]
Comment (pyrofolium) [reply cardinal/ecs/gamestate/ecb.go:106]: The error is already wrapped by the internal functions that return error with eris, which will return the stack trace.

File: cardinal/ecs/gamestate/gamestatemanager.go
Comment (technicallyty) [inline cardinal/ecs/gamestate/gamestatemanager.go:1]: lets rename this to `manager.go`

Comment (pyrofolium) [reply cardinal/ecs/gamestate/gamestatemanager.go:1]: Done @technicallyty

File: cardinal/ecs/gamestate/ecb_test.go [hunk 6]
```diff
@@ -19,19 +21,21 @@ import (
 	"github.com/redis/go-redis/v9"
 
 	"pkg.world.dev/world-engine/cardinal/ecs"
-	"pkg.world.dev/world-engine/cardinal/ecs/ecb"
 	"pkg.world.dev/world-engine/cardinal/types/component"
 	"pkg.world.dev/world-engine/cardinal/types/entity"
 )
 
-func newCmdBufferForTest(t *testing.T) *ecb.Manager {
+func newCmdBufferForTest(t *testing.T) *gamestate.EntityComponentBuffer {
```
Comment (jerargus) [inline cardinal/ecs/gamestate/ecb_test.go:28]: There's a mix of CommandBuffer and ComponentBuffer in this PR. It should all be CommandBuffer or CmdBuffer.

Comment (pyrofolium) [reply cardinal/ecs/gamestate/ecb_test.go]: @jerargus fixed

**Review by jerargus — APPROVED** [review]

**Review by namrapatel — APPROVED** [review]

**Pull request merged**